csharp/src/Ice/ConnectionI.cs

```diff
@@ -1169,14 +1169,8 @@ private void finish()
                 s.Append(" connection\n");
                 s.Append(ToString());
 
-                //
-                // Trace the cause of unexpected connection closures
-                //
-                if (!(_exception is CloseConnectionException ||
-                     _exception is ConnectionAbortedException ||
-                     _exception is ConnectionClosedException ||
-                     _exception is CommunicatorDestroyedException ||
-                     _exception is ObjectAdapterDeactivatedException))
+                // Trace the cause of most connection closures.
+                if (!(_exception is CommunicatorDestroyedException || _exception is ObjectAdapterDeactivatedException))
```

Comment:
We're now giving more details for:
 - CloseConnectionException (i.e. closed by peer)
 - ConnectionAbortedException (i.e. aborted by idle timer or by application)
 - ConnectionClosedException (i.e. closed by application or by inactivity check)